tests/client/amo/components/TestAddonReview.js

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import { findDOMNode } from 'react-dom';
+import {
+  findRenderedComponentWithType,
+  renderIntoDocument,
+  Simulate,
+} from 'react-addons-test-utils';
+
+import { setReview } from 'amo/actions/reviews';
+import * as amoApi from 'amo/api';
+import * as coreApi from 'core/api';
+import {
+  mapDispatchToProps, mapStateToProps, AddonReviewBase,
+  loadAddonReview as defaultAddonReviewLoader,
+} from 'amo/components/AddonReview';
+import I18nProvider from 'core/i18n/Provider';
+import { fakeAddon, signedInApiState } from 'tests/client/amo/helpers';
+import { getFakeI18nInst, RouterStub } from 'tests/client/helpers';
+
+const defaultReview = {
+  id: 3321, addonSlug: fakeAddon.slug,
+};
+
+function render({ fakeRouter = {}, ...customProps } = {}) {
+  const props = {
+    i18n: getFakeI18nInst(),
+    apiState: signedInApiState,
+    review: defaultReview,
+    updateReviewText: () => {},
+    ...customProps,
+  };
+  const root = findRenderedComponentWithType(renderIntoDocument(
+    <I18nProvider i18n={props.i18n}>
+      <RouterStub router={fakeRouter}>
+        <AddonReviewBase {...props} />
+      </RouterStub>
+    </I18nProvider>
+  ), AddonReviewBase);
+
+  return findDOMNode(root);
+}
+
+describe('AddonReview', () => {
+  it('can update a review', () => {
+    const fakeRouter = {};
+    const updateReviewText = sinon.spy(() => {});
+    const rootNode = render({ updateReviewText, fakeRouter });
+
+    const textarea = rootNode.querySelector('textarea');
```

Comment:
We've been using refs in some other tests to find elements instead of using `querySelector`. Seems like it should be a little less fragile that way.